hotshot-state-prover/src/service.rs

```diff
@@ -54,7 +56,19 @@ use crate::snark::{generate_state_update_proof, Proof, ProvingKey};
 type F = ark_ed_on_bn254::Fq;
 
 /// A wallet with local signer and connected to network via http
-pub type L1Wallet = SignerMiddleware<Provider<Http>, LocalWallet>;
+pub type SignerWallet = SignerMiddleware<Provider<Http>, LocalWallet>;
+
+#[derive(Debug, Clone)]
+pub struct AltChainConfig {
```

Comment:
Currently I'm not sure if handling multiple chains in the same prover service is advantageous. I think we could run multiple prover services for multiple chains. Or does that create any issues?

With multiple prover services they can be configured independently and use different intervals for different chains which is somewhat important because the gas cost can vary a lot between chains.

Reply:
makes sense! I have pushed the changes